File: app/src/test/java/com/example/ecgreb/mvpc/model/AddingMachineTest.java
package com.example.ecgreb.mvpc.model;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class AddingMachineTest {
  private AddingMachine addingMachine = new AddingMachine();

  @Test public void shouldNotBeNull() throws Exception {
    assertThat(addingMachine).isNotNull();
  }

  @Test public void sum_shouldReturnNullForNullAddend()
      throws Exception {
    assertThat(addingMachine.sum(null, null)).isNull();
  }

  @Test public void sum_shouldReturnNullForEmptyAddend()
      throws Exception {
    assertThat(addingMachine.sum("", "")).isNull();
  }

  @Test public void sum_shouldReturnNullForNonNumberAddend()
      throws Exception {
    assertThat(addingMachine.sum("a", "b")).isNull();
  }

  @Test public void sum_shouldReturnCorrectResultAsString() throws Exception {
    assertThat(addingMachine.sum("2", "3")).isEqualTo("5");
  }
}

File: app/src/test/java/com/example/ecgreb/mvpc/model/AddingMachineTest.kt
package com.example.ecgreb.mvpc.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class AddingMachineTest {
  private val addingMachine = AddingMachine()

  @Test fun shouldNotBeNull() {
    assertThat(addingMachine).isNotNull()
  }

  @Test fun sum_shouldReturnNullForNullAddend() {
    assertThat(addingMachine.sum(null, null)).isNull()
  }

  @Test fun sum_shouldReturnNullForEmptyAddend() {
    assertThat(addingMachine.sum("", "")).isNull()
  }

  @Test fun sum_shouldReturnNullForNonNumberAddend() {
    assertThat(addingMachine.sum("a", "b")).isNull()
  }

  @Test fun sum_shouldReturnCorrectResultAsString() {
    assertThat(addingMachine.sum("2", "3")).isEqualTo("5")
  }
}
